crates/mun_runtime/src/struct.rs

```diff
@@ -155,7 +176,9 @@ impl ArgumentReflection for StructRef {
     type Marshalled = RawStruct;
 
     fn type_name(&self) -> &str {
-        self.info.name()
+        let runtime_ref = self.runtime.borrow();
+        let type_info = unsafe { &*runtime_ref.gc().ptr_type(self.handle.handle()).inner() };
+        type_info.name()
```

Comment:
This in particular worries me, because there is a significant performance overhead to retrieving a struct's type information. This change affects `StructRef::get`, `StructRef::replace`, and `invoke_fn!`.

It would already be a lot better if we didn't still have to `borrow` the `Runtime`. That means passing the `&Runtime` as an additional parameter to the `ArgumentReflection` trait's `type_name` and `type_guid` functions, though.